### PR TITLE
Staging & CI support

### PR DESCRIPTION
**Pull request by jgadling**
### Description

This adds terraform code and happy CLI configuration to enable deploying Aspen to the staging environment.
We're combining ts-tests and py-tests into the same workflow, and deploying to the staging environment (commented out until after our upcoming demo) after all tests succeed.

There's only one minor change to the application - the current staging task definition sets REMOTE_DEV_ENV to empty string ("") instead of not setting it at all, which prevents us from using the default ("aspen_db") database name. We now ignore an empty string in the staging env.

#### Issue
[ch<fill_in_issue_number>](https://app.clubhouse.io/genepi/story/<fill_in_issue_number>)

### Test plan

Write how your changes are tested, or give a convincing reason why they can't be tested automatically.

### CODE REVIEW

File: src/py/aspen/config/config.py [hunk 1]
```diff
@@ -244,5 +244,5 @@ def DATABASE_URI(self) -> str:
             instance_address = instance_info["Endpoint"]["Address"]
             instance_port = instance_info["Endpoint"]["Port"]
             instance_address = f"{instance_address}:{instance_port}"
-        db_name = os.getenv("REMOTE_DEV_PREFIX", "/aspen_db")
+        db_name = os.getenv("REMOTE_DEV_PREFIX") or "/aspen_db"
```
Comment (jgadling) [inline src/py/aspen/config/config.py:247]: This is a workaround to make sure we use "aspen_db" if the REMOTE_DEV_PREFIX env var is an empty string.

File: .github/workflows/push-tests.yml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.ECR_REPO }}
+      - uses: actions/checkout@v2
+      - name: Docker build, push, and tag
+        shell: bash
+        run: |
+          pip install -r .happy/requirements.txt
+          scripts/happy --profile="" addtags --source-tag sha-${GITHUB_SHA:0:8} --dest-tag branch-$(echo ${GITHUB_REF#refs/heads/} | sed 's/[\+\/]/-/g')
```
Comment (jgadling) [inline .github/workflows/push-tests.yml:153]: Adds the "branch-trunk" tag to this build on success

File: .github/workflows/deploy-happy-stack.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        with:
+          state: in_progress
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Update deployment
+        env:
+          TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
+        run: |
+          # TODO - staging deploy is disabled until after Tues demo!!
+          # ./scripts/happy --profile="" --env ${{ github.event.deployment.environment }} update ${{ github.event.deployment.environment }}stack --tag ${{ github.event.deployment.payload.tag }}
```
Comment (jgadling) [inline .github/workflows/deploy-happy-stack.yml]: This is commented out for now, we can uncomment it after our demo!

File: .github/workflows/push-tests.yml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+          role-duration-seconds: 900
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.ECR_REPO }}
+      - uses: actions/checkout@v2
+      - name: Test ts build
+        run: |
+          echo "DOCKER_REPO=${DOCKER_REPO}" > .env.ecr
+          mkdir src/ts/build
```
Comment (phoenixAja) [inline .github/workflows/push-tests.yml:77]: do we still need this fix now that we added build to .dockerignore?

**Review by phoenixAja — APPROVED** [review]

**Comment by phoenixAja** [comment]
looks good!! exciting!

File: .github/workflows/push-tests.yml (hunk 4 above)
Comment (jgadling) [reply .github/workflows/push-tests.yml:77]: errr ... you might be right. let me look into that!

**Pull request merged**